src/main/java/com/ait/lienzo/client/core/shape/wires/layout/direction/HorizontalLayoutFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.wires.layout.direction;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.ait.lienzo.client.core.shape.wires.layout.direction.DirectionLayout.Direction;
+import com.ait.lienzo.client.core.shape.wires.layout.direction.DirectionLayout.HorizontalAlignment;
+import com.ait.lienzo.client.core.shape.wires.layout.direction.DirectionLayout.Orientation;
+import com.ait.lienzo.client.core.shape.wires.layout.direction.DirectionLayout.ReferencePosition;
+import com.ait.lienzo.client.core.types.BoundingBox;
+import com.ait.tooling.common.api.java.util.function.Function;
+
+public class HorizontalLayoutFactory
+{
+
+    private static Map<ReferencePosition, DirectionLayoutBuilder<HorizontalAlignment>> builders = new HashMap<ReferencePosition, DirectionLayoutBuilder<HorizontalAlignment>>()
+    {{
```

Comment:
mmm I remember @hasys refactored the inline instance initialization from the use of the double-brace syntax to the use of some builder classes... can you please @hasys explain (again, sorry) why it's better not to use this syntax?

Reply:
Hi @tiagodolphine, @romartin, yes, it creates excessive anonymous classes. I guess we can't use builder from https://github.com/kiegroup/kie-soup/commit/f74f7fd156f7fb425ec89e43ee55adac1fae85a1#diff-aa75dec6ef10c21c3d2203da2d7f5e54 due to new dependency, but we can use normal initialization and #add method.

See discussion from this PR for more details: https://github.com/kiegroup/kie-wb-common/pull/1971

Reply:
yes I remember and in fact I'm using on the kie-wb-common, but on lienzo for now we don't have this builder, since this is a final static attribute, just instantiated once, I chose this initialization, but I can replace with a method, np

Reply:
done